examples/animation/src/main.rs

```diff
@@ -0,0 +1,126 @@
+#![cfg_attr(not(debug_assertions), windows_subsystem = "windows")] // hide console window on Windows in release
+
+use eframe::egui;
+use egui::{color, Color32, Id, Rect, Rounding};
+use std::ops::Range;
+
+fn main() {
+    let options = eframe::NativeOptions::default();
+    eframe::run_native(
+        "My egui App",
+        options,
+        Box::new(|cc| {
+            let app = MyApp::default();
+            // Set the initial opacity of the rectangle to write the current value to memory in
+            // the animation manager.
+            cc.egui_ctx
+                .animate_value_with_time(app.rectangle_id, app.current_opacity, 0.0);
+            Box::new(app)
+        }),
+    );
+}
+
+#[derive(PartialEq)]
+enum AnimateDirection {
+    FadeIn,
+    FadeOut,
+}
+
+struct MyApp {
+    rectangle_id: Id,
+    opacity_range: Range<f32>,
+    animate_direction: AnimateDirection,
+    fade_in_time: f32,
+    fade_out_time: f32,
+    current_opacity: f32,
```

Comment:
The whole point of the egui animation system is so you don't need to keep this much state around. Really, for you example, you should only need a single `bool`.

The `toggle_switch.rs` example demonstrates this pretty well: https://github.com/emilk/egui/blob/master/crates/egui_demo_lib/src/demo/toggle_switch.rs#L49